js/classes/status_effect.js

```diff
@@ -0,0 +1,395 @@
+                }
+                break;
+
+            case StatusEffect.FAST:
+                this.fast = false;
+                this.entity.speed = this.defaultSpeed;
+
+                if (this.entity === CHAD) {
+                    this.entity.firstJumpVelocity = Chad.DEFAULT_FIRST_JUMP_VELOCITY;
+                    this.entity.secondJumpVelocity = Chad.DEFAULT_SECOND_JUMP_VELOCITY;
+                }
+                break;
+
+            case StatusEffect.SLOW:
+                this.slow = false;
+                this.entity.speed = this.defaultSpeed;
+                break;
+
+            case StatusEffect.GIANT:
+                this.giant = false;
+                this.restoreScale(); // this applies to all entities, not just Chad
+                break;
+
+            default:
+                throw new Error("Invalid status effect: " + effect);
+        }
+    }
+
+
+    /**
+     * Update the status effect, decreasing timers and removing effects when they expire.
+     */
+    update() {
+        if (this.invincible) {
+            this.invincibleTimer -= GAME.clockTick;
+            if (GAME.gameTime % 0.1 < 0.01) {
+                GAME.addEntity(new ParticleEffect(
+                    this.entity.getCenter(),
+                    ParticleEffect.GOLD_SPARKLE)
+                );
+            }
+            if (this.invincibleTimer <= 0) {
+                this.remove(StatusEffect.INVINCIBLE);
+            }
+        }
+
+        if (this.strong) {
+            this.strongTimer -= GAME.clockTick;
+            if (GAME.gameTime % 0.1 < 0.01) {
+                GAME.addEntity(new ParticleEffect(
+                    this.entity.getCenter(),
+                    ParticleEffect.RED_SPARKLE)
+                );
+            }
+            if (this.strongTimer <= 0) {
+                this.remove(StatusEffect.STRONG);
+            }
+        }
+
+        if (this.fast) {
+            this.fastTimer -= GAME.clockTick;
+            if (GAME.gameTime % 0.1 < 0.01) {
+                GAME.addEntity(new ParticleEffect(
+                    this.entity.getCenter(),
+                    ParticleEffect.GREEN_SPARKLE)
+                );
+            }
+            if (this.fastTimer <= 0) {
+                this.remove(StatusEffect.FAST);
+            }
+        }
+
+        if (this.slow) {
+            this.slowTimer -= GAME.clockTick;
+            if (GAME.gameTime % 0.1 < 0.01) {
+                GAME.addEntity(new ParticleEffect(
+                    this.entity.getCenter(),
+                    ParticleEffect.GRAY_SPARKLE)
+                );
+            }
+            if (this.slowTimer <= 0) {
+                this.remove(StatusEffect.SLOW);
+            }   
+        }
+
+        if (this.giant) {
+            this.giantTimer -= GAME.clockTick;
+            if (this.giantTimer <= 0) {
+                this.remove(StatusEffect.GIANT);
+            }
+            if (this.canCrushTimer > 0) {
```

Comment:
I'm confused at what a can Crush Timer is.

Reply:
Oh I see. Can we consider a rename to stompTimer or crushTimer? Not a huge deal, just had 'can' in my mind due to energy drink so I thought it was something else.